sos/report/plugins/ceph_mon.py

```diff
@@ -88,63 +135,96 @@ def setup(self):
-                    try:
-                        id_index = proc.index("--id")
-                        mon_ids.append(proc[id_index+1])
-                    except (IndexError, ValueError):
-                        self.log_warn("could not find ceph-mon id: %s", procs)
-
-        self.add_cmd_output([
-            "ceph tell mon.%s mon_status" % mon_id for mon_id in mon_ids
-        ], subdir="json_output", tags="insights_ceph_health_detail")
-
-        self.add_forbidden_path([
-            "/etc/ceph/*keyring*",
-            "/var/lib/ceph/*keyring*",
-            "/var/lib/ceph/*/*keyring*",
-            "/var/lib/ceph/*/*/*keyring*",
-            # Excludes temporary ceph-osd mount location like
-            # /var/lib/ceph/tmp/mnt.XXXX from sos collection.
-            "/var/lib/ceph/tmp/*mnt*",
-            "/etc/ceph/*bindpass*"
-        ])
+        self.add_cmd_output(
+            [f"ceph tell mon.{mid} mon_status" for mid in self.get_ceph_ids()],
+            subdir="json_output", tags="insights_ceph_health_detail"
+        )
 
-        # If containerized, run commands in containers
-        try:
-            cname = self.get_all_containers_by_regex("ceph-mon*")[0][1]
-        except Exception:
-            cname = None
+        self.add_cmd_output([f"ceph {cmd}" for cmd in ceph_cmds])
 
+        # get ceph_cmds again as json for easier automation parsing
         self.add_cmd_output(
-            ["ceph %s" % cmd for cmd in ceph_cmds],
-            container=cname
+            [f"ceph {cmd} --format json-pretty" for cmd in ceph_cmds],
+            subdir="json_output", tags="insights_ceph_health_detail"
         )
 
+    def get_ceph_version(self):
+        ver = self.exec_cmd('ceph --version')
+        if ver['status'] == 0:
+            try:
+                _ver = ver['output'].split()[2]
+                return int(_ver.split('.')[0])
+            except Exception as err:
+                self._log_debug(f"Could not determine ceph version: {err}")
+        self._log_error(
+            'Failed to find ceph version, command collection will be limited'
+        )
+        return 0
+
+    def get_ceph_ids(self):
+        ceph_ids = []
+        # ceph version 14 correlates to RHCS 4
+        if self.ceph_version == 14:
+            # Get the ceph user processes
+            out = self.exec_cmd('ps -u ceph -o args')
+
+            if out['status'] == 0:
+                # Extract the mon ids
+                for procs in out['output'].splitlines():
+                    proc = procs.split()
+                    # Locate the '--id' value of the process
+                    if proc and proc[0].endswith("ceph-mon"):
+                        try:
+                            id_index = proc.index("--id")
+                            ceph_ids.append(proc[id_index + 1])
+                        except (IndexError, ValueError):
+                            self._log_warn('Unable to find ceph IDs')
+
+        # ceph version 16 is RHCS 5
```

Comment:
I think this is very RH specific. Also, what happens for ceph version 15, octopus? The if condition in this function only covers version == 14, or >=16..